cloud_utils/scheduler/kubernetes_connector.py

```diff
@@ -134,6 +134,9 @@ def create_cron_job(
 def _make_pod_manifest(
     env_variables: List[Dict[Text, Text]],
     secrets: List[Dict[Text, Text]],
+    provide_secrets: List[str],
+    secret_provider_class: str,
+    resources: Dict[str, Dict[str, str]],
```

Comment:
typing should be Optional (according to the usage in the function)

Reply:
if it's an emptiness check, prefer gamla.empty() / gamla.nonempty()

Reply:
Changed to be non-optional